server/lib/shared.py

```diff
@@ -19,22 +19,41 @@
 import server.lib.fetch as fetch
 
 
-def names(dcids):
+def names(dcids, prop=None):
   """Returns display names for set of dcids.
 
   Args:
       dcids: A list of DCIDs.
+      prop: property to use to get the names.
 
   Returns:
       A dictionary of display place names, keyed by dcid.
   """
-  response = fetch.property_values(dcids, 'name')
   result = {}
-  for dcid in dcids:
-    result[dcid] = ''
-    for dcid, values in response.items():
-      if values:
-        result[dcid] = values[0]
+  # dcids to get default name for
+  default_name_dcids = []
+  # if there's a specified property to use for the name, try to find names using
+  # that prop first
+  if prop:
+    name_prop_response = fetch.property_values(dcids, prop)
+    for dcid in dcids:
+      for dcid, values in name_prop_response.items():
+        name_found = False
```

Comment:
There two variables called dcid in the same scope?

Maybe you want:

```
for dcid in dcids:
  if resp.get(dcid, []):
    result[dcid] = resp[dcid][0]
  else:
     default_name_dcids.append(dcid)  
```

ditto, below

Reply:
updated